core/core.go

```diff
@@ -87,8 +93,10 @@ type NetworkDriver interface {
 	DeleteNetwork(id string) error
 	CreateEndpoint(id string) error
 	DeleteEndpoint(id string) error
-	CreatePeerHost(id string) error
-	DeletePeerHost(id string) error
+	AddPeerHost(node ServiceInfo) error
```

Comment:
I did embed the clustering APIs in NetworkDriver interface. I converted 'PeerHost' interface to 'Clustering' interface for plugin interface. Similar to how there is separate 'Network' and 'Endpoint' interfaces.